libcxx/modules/modules.json.in

```diff
@@ -0,0 +1,26 @@
+{
+  "version": 1,
+  "revision": 1,
+  "modules": [
+    {
+      "logical-name": "std",
+      "source-path": "@LIBCXX_MODULE_RELATIVE_PATH@/std.cppm",
+      "is-standard-library": true,
+      "local-arguments": {
+        "system-include-directories": [
+          "@LIBCXX_MODULE_RELATIVE_PATH@"
+        ]
+      }
+    },
+    {
+      "logical-name": "std.compat",
+      "source-path": "@LIBCXX_MODULE_RELATIVE_PATH@/std.compat.cppm",
+      "is-std-library": true,
```

Comment:
This isn't consistent with `is-standard-library` above. Which is it?

Reply:
Could we please use valid python identifiers for keys? That way we can load the json into typed models.

Reply:
Or has that ship sailed?

Reply:
FWIW, [P1689](https://www.open-std.org/jtc1/sc22/wg21/docs/papers/2022/p1689r5.html) uses kebab-case as well. Note that 18 is in rc3; I don't know how much "Python APIs are easier" is to make a release blocker versus "the file is not consistent with itself". Something for release managers to decide.

Note that there's nothing stopping a typed model from acting on it as-is because a function can transform these into identifier names (IME, the explicitness from things like `#[serde(rename = "…")]` are better than forcing the interchange format to bend to a specific language's ergonomics).

Reply:
Sure. I agree that this probably isn't important enough to do anything about. And standardizing behind a specific format is more important.

But I also don't think it's unreasonable to prefer interacting with the data in a typed/schema-enforced way, and that treating keys as identifiers rather than strings is preferable to achieve that.

However, if they're no tooling that yet depends on the format, I think it's worth changing. Because consistency isn't everything, but it's nice to have, and we might lose the ability to have it later.

That said, this _can_ land after the release, it's just a little uglier of a scenario.

Reply:
I should be clear, I'm happy to make the changes myself. But I wanted to make sure there was at least tacit approval .

Reply:
I've no strong opinion about the naming. However as mentioned in the commit message the naming is based on a discussion in SG15 and as @mathstuf mentioned is used in P1689. (clang-scan-deps and CMake (and probably others too) have implemented this paper.) So IMO changing the naming should be discussed in SG15.

Reply:
> This isn't consistent with `is-standard-library` above. Which is it?

Thanks for catching this. I should have used copy-paste programming ;-)

Reply:
> Thanks for catching this. I should have used copy-paste programming ;-)

No problem. Was very confused when "no such key" errors came up. Changed it to the one in the line it complained about…still errored. Finally saw the difference. Just glad I got around to testing this before the final release.

Reply:
btw does that mean you're working on implementing this is CMake?

Reply:
Yep :) .